aws_lambda_powertools/metrics/provider/datadog/datadog.py

```diff
@@ -0,0 +1,291 @@
+from __future__ import annotations
+
+import json
+import logging
+import numbers
+import os
+import time
+import warnings
+from typing import Any, Callable, Dict, List, Optional
+
+from aws_lambda_powertools.metrics.exceptions import MetricValueError, SchemaValidationError
+from aws_lambda_powertools.metrics.provider import BaseProvider
+from aws_lambda_powertools.shared import constants
+from aws_lambda_powertools.shared.functions import resolve_env_var_choice
+from aws_lambda_powertools.utilities.typing import LambdaContext
+
+logger = logging.getLogger(__name__)
+
+# Check if using datadog layer
+try:
+    from datadog_lambda.metric import lambda_metric  # type: ignore
+except ImportError:  # pragma: no cover
+    lambda_metric = None  # pragma: no cover
+
+DEFAULT_NAMESPACE = "default"
+
+
+class DatadogProvider(BaseProvider):
+    """
+    DatadogProvider creates metrics asynchronously via Datadog extension or exporter.
+
+    **Use `aws_lambda_powertools.DatadogMetrics` to create and metrics to Datadog.**
+
+    Environment variables
+    ---------------------
+    POWERTOOLS_METRICS_NAMESPACE : str
+        metric namespace to be set for all metrics
+
+    Raises
+    ------
+    MetricValueError
+        When metric value isn't a number
+    SchemaValidationError
+        When metric object fails EMF schema validation
+    """
+
+    def __init__(
+        self,
+        metric_set: List | None = None,
+        namespace: str | None = None,
+        flush_to_log: bool | None = None,
+        default_tags: List | None = None,
+    ):
+        self.metric_set = metric_set if metric_set is not None else []
+        self.namespace = resolve_env_var_choice(choice=namespace, env=os.getenv(constants.METRICS_NAMESPACE_ENV))
+        if self.namespace is None:
+            self.namespace = DEFAULT_NAMESPACE
+        self.default_tags = default_tags or []
+        self.flush_to_log = resolve_env_var_choice(choice=flush_to_log, env=os.getenv(constants.DATADOG_FLUSH_TO_LOG))
+
+    #  adding name,value,timestamp,tags
+    def add_metric(
+        self,
+        name: str,
+        value: float,
+        timestamp: int | None = None,
+        tags: List | None = None,
+        **kwargs: Any,
```

Comment:
you can improve the experience by using a `**tags` instead. 

`add_metric(..., tags=["a":"b"])`

**vs**

`add_metric(..., "a"="b")`

It'll also make your logic more nimble too when unpacking them.

Reply:
Yep, make more sense. Done.